website/docs/language/meta-arguments/lifecycle.mdx

```diff
@@ -49,6 +49,14 @@ The arguments available within a `lifecycle` block are `create_before_destroy`,
   such features, so you must understand the constraints for each resource
   type before using `create_before_destroy` with it.
 
+  It's important to understand that `create_before_destroy` meta-attribute is
+  stored in state file and _propagated_ to all resource dependencies, including
+  transitive. For example, resource A depends on resource B, hence B can't be
+  destroyed prior to A. The new A depends on the new B, so that has to be made
+  first, hence forcing the B to be `create_before_destroy` as well. You can't
+  override `create_before_destroy` to `false` on B, because the ordering of
+  resource requires it, otherwise there would be dependency cycles in the graph.
+
```

Comment:
```suggestion
  Note that Terraform stores the `create_before_destroy` meta-attribute in the state file that applies to all resource dependencies. The attribute is transitive, so dependencies for resources with `create_before_destroy` enabled also inherit the `create_before_destroy` behavior. For example, if resource A depends on resource B, then Terraform cannot destroy resource B prior to destroying resource A. If the `create_before_destroy` meta-attribute is enabled for resource A and Terraform creates new instances of resources A and B, then Terraform cannot destroy the previous instance of resource B until it destroys the previous instance of resource A. You cannot
  override `create_before_destroy` to `false` on B because the order of
  resource creation and destruction requires it. Otherwise, there would be dependency cycles in the graph.

```
I tried to make the explanation more explicit. As is, we're implying the behavior through an example. Please let me know if I misunderstood and changed the meaning.

Reply:
Well, I'm not native English speaking person, but **propagation** feels like a more correct term reckoning my university course 😅 

**Attribute propagation** involves the transfer or propagation (verb) of attributes from one entity to another within a system. It is a process where an attribute of an entity is **passed or copied** to another object or entity. The goal of attribute propagation is to ensure that relevant information or characteristics are transmitted and maintained across different components or entities. That's exactly what I observed Terraform has done – copied `create_before_destroy` to all objects in the chain.

**Attribute transitive** refers to the transitive property (noun) of attributes in a system. Transitivity means that if attribute A is related to attribute B, and attribute B is related to attribute C, then attribute A can be considered indirectly related to attribute C. In other words, the transitive property allows attributes to be inherited or derived through a chain of relationships.

Anyway, I'm OK with your variant. The most important – pay attention of a regular user to this intrinsic behaviour.

Reply:
Thanks for the explanation, @Tensho 
Do you mind committing the suggestion if it makes sense so that we can merge this PR? We appreciate your help!

Reply:
Done